frontend/sozisel-app/src/components/PresenterSession/EventsTimline/EventsTimeline.tsx

```diff
@@ -0,0 +1,220 @@
+}
+
+interface EventsTimelineProps {
+  events: Event[];
+  launchedEvents: LaunchedEvent[];
+  sessionId: string;
+  participants: Participant[];
+}
+
+interface TimerProps {
+  onFinishCallback: () => void;
+  startValue: number;
+}
+
+const Timer = ({ onFinishCallback, startValue }: TimerProps): ReactElement => {
+  const countdownTimer = useCountdownTimer({
+    startValue,
+    onFinishCallback,
+  });
+
+  return <p className="Timer">{countdownTimer}</p>;
+};
+
+const getRandomParticipants = (
+  participants: Participant[],
+  targetPercentageOfParticipants: number
+): string[] => {
+  return participants
+    .sort(() => 0.5 - Math.random())
+    .slice(
+      0,
+      Math.floor((targetPercentageOfParticipants / 100) * participants.length)
+    )
+    .map((participant) => participant.id);
+};
+
+export default function EventsTimeline({
+  events,
+  launchedEvents,
+  sessionId,
+  participants,
+}: EventsTimelineProps): ReactElement {
+  const { t } = useTranslation("common");
+  const [activeEventIdx, setActiveEventIdx] = useState<number>(0);
+  const [activeEventId, setActiveEventId] = useState<string>("");
+  const [activeEventCurrentSec, setActiveEventCurrentSec] = useState<number>(0);
+  const history = useHistory();
+
+  const [endSessionMutation, { error: endSessionError }] =
+    useEndSessionMutation();
+
+  const onEndSession = async () => {
+    await endSessionMutation({ variables: { id: sessionId } });
+    if (!endSessionError) {
+      history.push(`result/summary`);
+    }
+  };
+
+  useEffect(() => {
+    const lastEventIdx = launchedEvents.length - 1;
+    if (lastEventIdx >= 0) {
+      const lastEvent = launchedEvents[launchedEvents.length - 1];
+      const lastEventStartTime = moment.tz(
+        lastEvent.insertedAt,
+        "Europe/Warsaw"
+      );
+      const currentTime = Math.floor(
+        moment
+          .duration(moment.tz("Europe/Warsaw").diff(lastEventStartTime))
+          .asSeconds()
+      );
+      if (currentTime < events[lastEventIdx].eventData.durationTimeSec) {
+        setActiveEventIdx(lastEventIdx);
+        setActiveEventId(lastEvent.id);
+        setActiveEventCurrentSec(
+          events[lastEventIdx].eventData.durationTimeSec - currentTime
+        );
+      } else {
+        setActiveEventIdx(lastEventIdx + 1);
+      }
+    }
+  }, [activeEventIdx, events, launchedEvents]);
+
+  const [launchEventMutation, { error: launchEventError }] =
+    useLaunchEventMutation();
+
+  const onNextEvent = async () => {
+    const {
+      id,
+      eventData: { targetPercentageOfParticipants },
+    } = events[activeEventIdx];
+    await launchEventMutation({
+      variables: {
+        broadcast: targetPercentageOfParticipants === 100,
+        eventId: id,
+        sessionId,
+        targetParticipants: getRandomParticipants(
+          participants,
+          targetPercentageOfParticipants
+        ),
+      },
+    });
+    if (!launchEventError && activeEventIdx < events.length) {
```

Comment:
Don't use errors like that, it doesn't work like that whatsoever... It you are doing the await you have to catch the error if you want to use the errors immediately. Either do the try catch or  don't do the await and do the `useEffect` on the error values that will come. The second approach is different from the first one as the later will print an error to the console while doing try/catch will not.